From 631b4d062748410858bc873ad65e145e58dfe4e9 Mon Sep 17 00:00:00 2001
From: Ariadne Conill <ariadne@dereferenced.org>
Date: Tue, 18 Feb 2020 21:10:22 +0000
Subject: [PATCH] powerpc: config: defang gcc check for stack-protector support

When we rebase kernel configs in Alpine, it is common to do so from
an environment where CARCH does not match CHOST.

Accordingly, we assume that the gcc check will always pass, and
simply remove it in preference for always enabling the
HAVE_STACKPROTECTOR config option.

Signed-off-by: Ariadne Conill <ariadne@dereferenced.org>
---
 arch/powerpc/Kconfig | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/powerpc/Kconfig b/arch/powerpc/Kconfig
index 2ca5418457ed..b62dab194e8e 100644
--- a/arch/powerpc/Kconfig
+++ b/arch/powerpc/Kconfig
@@ -248,8 +248,7 @@ config PPC
 	select HAVE_RSEQ
 	select HAVE_SETUP_PER_CPU_AREA		if PPC64
 	select HAVE_SOFTIRQ_ON_OWN_STACK
-	select HAVE_STACKPROTECTOR		if PPC32 && $(cc-option,-mstack-protector-guard=tls -mstack-protector-guard-reg=r2)
-	select HAVE_STACKPROTECTOR		if PPC64 && $(cc-option,-mstack-protector-guard=tls -mstack-protector-guard-reg=r13)
+	select HAVE_STACKPROTECTOR
 	select HAVE_STATIC_CALL			if PPC32
 	select HAVE_SYSCALL_TRACEPOINTS
 	select HAVE_VIRT_CPU_ACCOUNTING
-- 
2.38.2

